Sources/SwiftConcurrentSequence/Sequence+ConcurrentReduce.swift

```diff
@@ -0,0 +1,187 @@
+                DispatchQueue.concurrentPerform(iterations: toReduce.count) { index in
+                    reducingIntoFirst(&bufferPointer[index], toReduce[index])
+                }
+            }
+        }
+
+        return reduced.first ?? defaultValue()
+    }
+
+    /// Returns the result of combining the elements of the sequence of dictionaries
+    /// using the given closure. The given closure is executed concurrently
+    /// on multiple queues to reduce the wall-time consumed by the reduction.
+    ///
+    /// Use the `reduce(into:_:)` method to produce a single value from the
+    /// elements of an entire sequence. For example, you can use this method on an
+    /// array of integers to filter adjacent equal entries or count frequencies.
+    ///
+    /// - Parameters:
+    ///   - defaultValue: A default value for Element. This value is utilized only
+    ///     when the receiver is empty.
+    ///   - mergeConflictResolver: A closure that returns the desired value for
+    ///     the given key when multiple values are present for the given key.
+    /// - Returns: The final reduced value. If the sequence has no elements,
+    ///   the result is `defaultValue`.
+    public func concurrentReduce<Key, Value>(
+        mergeConflictResolver: (Key, Value, Value) -> Value)
+    -> Element
+    where Element == [Key: Value]
+    {
+        concurrentReduce(defaultValue: Element()) { updating, next in
+            for (key, nextValue) in next {
+                if let existingValue = updating[key] {
+                    updating[key] = mergeConflictResolver(key, existingValue, nextValue)
+                } else {
+                    updating[key] = nextValue
+                }
+            }
+        }
+    }
+
+    /// Returns the result of combining the elements of the sequence using the
+    /// given closure. The given closure is executed concurrently
+    /// on multiple queues to reduce the wall-time consumed by the reduction.
+    ///
+    /// Use the `reduce(into:_:)` method to produce a single value from the
+    /// elements of an entire sequence. For example, you can use this method on an
+    /// array of integers to filter adjacent equal entries or count frequencies.
+    ///
+    /// - Parameters:
+    ///   - defaultValue: A default value for Element. This value is utilized only
+    ///     when the receiver is empty.
+    ///   - reducingIntoFirst: A closure that combines an accumulating value and
+    ///     an element of the sequence into a new reduced value, to be used
+    ///     in the next call of the `reducingIntoFirst` closure or returned to
+    ///     the caller.
+    /// - Returns: The final reduced value. If the sequence has no elements,
+    ///   the result is `defaultValue`.
+    public func concurrentReduce(
+        defaultValue: @escaping @autoclosure () -> Element,
+        _ reducer: @escaping (Element, Element) throws -> Element)
```

Comment:
note that our `reducer` in the `async` method here returns a new element. While it'd be nice to avoid copy-on-write when reducing structs, the compiler prevents me from sending an `inout` pointer across a structured concurrency barrier (and I don't blame it for doing so!).